src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/OrdinaryMethodReferenceFinder.cs

```diff
@@ -93,7 +93,11 @@ protected override bool CanFind(IMethodSymbol symbol)
                 ? await FindDocumentsWithForEachStatementsAsync(project, documents, cancellationToken).ConfigureAwait(false)
                 : ImmutableArray<Document>.Empty;
 
-            return ordinaryDocuments.Concat(forEachDocuments);
+            var deconstructDocuments = IsDeconstructMethod(methodSymbol)
```

Comment:
❓ Can/should we further narrow this down to cases where the parameter list only contains `out` parameters (or for static methods, only `out` parameters after a `this` parameter)?

Reply:
We could, but I'm not sure it's worth it. The purpose here is to prune potential documents and this simple heuristic is effective for that purpose.
For comparison, we apply a similar (simplistic) heuristic for detecting possible enumerator methods (see `IsForEachMethod` a few lines below), without checkout parameters or return type.
I feel that's ok to leave as-is.

---
In reply to: [161816905](https://github.com/dotnet/roslyn/pull/24223#discussion_r161816905) [](ancestors = 161816905)